mlflow/pyfunc/__init__.py

```diff
@@ -847,7 +848,23 @@ def predict(*args):
         else:
             return result[result.columns[0]]
 
-    return pandas_udf(predict, result_type)
+    udf = pandas_udf(predict, result_type)
+
+    @functools.wraps(udf)
+    def udf_with_default_cols(*args):
+        if len(args) == 0:
+            model = SparkModelCache.get_or_load(archive_path)
+            input_schema = model.metadata.get_input_schema()
+
+            if input_schema and input_schema.has_input_names():
+                input_names = input_schema.input_names()
+                return udf(*input_names)
```

Comment:
I think we should wrap the input_names in a struct for consistency, since that is what we should be recommending for users to do (e.g. in case the model has no signature). It does not really make a difference since we ask for the columns in the schema order and we autofill column names, but just in case there is any difference in behavior.

Reply:
I tried this out and haven't gotten it to work yet. Changing that line to `return udf(struct(*input_names))` results in this error:
```
TypeError: Unsupported type in conversion from Arrow: struct<a: int64, b: int64, c: int64>
```
I'm not sure if it's actually a bug, because the same error happens if I add `struct` wrappers to existing unit tests, but leave the code unchanged (master branch). See this example: https://github.com/mlflow/mlflow/pull/4244

Do you know how to avoid this issue when using structs, or should we leave the code as is?

Reply:
Oh I see, that's probably because we test with spark < 3. We can leave it as is then.